### ISSUE TITLE
[BUG] App Crashes on opening qr scanner on older devices

### ISSUE THREAD

**Issue by SushilGhorasaini1** [juliuscanute/qr_code_scanner#377]
While opening qr scanner on older devices having android version 5.1.1, app crashes with the error message:
E/flutter ( 5084): [ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: CameraException(cameraPermission, Platform Version to low for camera permission check)
E/flutter ( 5084): #0      QRViewController._startScan (package:qr_code_scanner/src/qr_code_scanner.dart:234)
E/flutter ( 5084): <asynchronous suspension>
E/flutter ( 5084):

**Flutter information**
[✓] Flutter (Channel stable, 2.2.3, on Linux, locale en_US.UTF-8)
    • Flutter version 2.2.3 at /home/sushil/snap/flutter/common/flutter
    • Framework revision f4abaa0735 (3 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/sushil/Android/Sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /snap/android-studio/113/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /snap/android-studio/113/android-studio
    • Flutter plugin version 58.0.1
    • Dart plugin version 202.8531
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] Connected device (2 available)
    • A37f (mobile) • 38cae8de • android-arm64  • Android 5.1.1 (API 22)
    • Chrome (web)  • chrome   • web-javascript • Google Chrome 92.0.4515.107

• No issues found!

**Additional context**
While opening app on phone having android version 11, it workes completely fine.
Previously, it was working completely fine on android v5.1.1 too

**Comment by juliansteenbakker** [comment]
I have found an issue with the code which just began to surface because the permission system has been changed. Can you please try again using the following branch and check if the problem is resolved: https://github.com/juliuscanute/qr_code_scanner/tree/bug/permission-android5-377

**Comment by SushilGhorasaini1** [comment]
Tried, But failed to get result. After scanning the QR code, App crashes with exception:
E/AndroidRuntime( 8699): FATAL EXCEPTION: DecoderThread
E/AndroidRuntime( 8699): Process: com.dashnetwork.dashfood, PID: 8699
E/AndroidRuntime( 8699): java.lang.NoSuchMethodError: No interface method sort(Ljava/util/Comparator;)V in class Ljava/util/List; or its super classes (declaration of 'java.util.List' appears in /system/framework/core-libart.jar)
E/AndroidRuntime( 8699):        at com.google.zxing.qrcode.detector.FinderPatternFinder.selectBestPatterns(FinderPatternFinder.java:624)
E/AndroidRuntime( 8699):        at com.google.zxing.qrcode.detector.FinderPatternFinder.find(FinderPatternFinder.java:164)
E/AndroidRuntime( 8699):        at com.google.zxing.qrcode.detector.Detector.detect(Detector.java:81)
E/AndroidRuntime( 8699):        at com.google.zxing.qrcode.QRCodeReader.decode(QRCodeReader.java:77)
E/AndroidRuntime( 8699):        at com.google.zxing.MultiFormatReader.decodeInternal(MultiFormatReader.java:173)
E/AndroidRuntime( 8699):        at com.google.zxing.MultiFormatReader.decodeWithState(MultiFormatReader.java:87)
E/AndroidRuntime( 8699):        at com.journeyapps.barcodescanner.Decoder.decode(Decoder.java:75)
E/AndroidRuntime( 8699):        at com.journeyapps.barcodescanner.Decoder.decode(Decoder.java:49)
E/AndroidRuntime( 8699):        at com.journeyapps.barcodescanner.DecoderThread.decode(DecoderThread.java:146)
E/AndroidRuntime( 8699):        at com.journeyapps.barcodescanner.DecoderThread.access$000(DecoderThread.java:22)
E/AndroidRuntime( 8699):        at com.journeyapps.barcodescanner.DecoderThread$1.handleMessage(DecoderThread.java:38)
E/AndroidRuntime( 8699):        at android.os.Handler.dispatchMessage(Handler.java:98)
E/AndroidRuntime( 8699):        at android.os.Looper.loop(Looper.java:160)
E/AndroidRuntime( 8699):        at android.os.HandlerThread.run(HandlerThread.java:61)
D/CameraPreview( 8699): pause()

**Comment by harlanx** [comment]
Having the same issue here.
It crashes immediately as soon as it recognizes the QR Code.
Android Version: 5.1.1

<details><summary>Logs</summary>
<p>

E/flutter (11283): [ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: CameraException(cameraPermission, Platform Version to low for camera permission check)
E/flutter (11283): #0      QRViewController._startScan (package:qr_code_scanner/src/qr_code_scanner.dart:234:7)
E/flutter (11283): <asynchronous suspension>
E/flutter (11283):
I/art     (11283): Background partial concurrent mark sweep GC freed 4158(303KB) AllocSpace objects, 12(7MB) 
LOS objects, 65% free, 4MB/12MB, paused 20.492ms total 38.623ms
I/art     (11283): Background sticky concurrent mark sweep GC freed 5665(426KB) AllocSpace objects, 13(5MB) LOS objects, 48% free, 6MB/11MB, paused 6.557ms total 16.159ms
E/AndroidRuntime(11283): FATAL EXCEPTION: DecoderThread
E/AndroidRuntime(11283): Process: com.example.qr_generator_reader, PID: 11283
E/AndroidRuntime(11283): java.lang.NoSuchMethodError: No interface method sort(Ljava/util/Comparator;)V in class Ljava/util/List; or its super classes (declaration of 'java.util.List' appears in /system/framework/core-libart.jar)
E/AndroidRuntime(11283):        at com.google.zxing.qrcode.detector.FinderPatternFinder.selectBestPatterns(FinderPatternFinder.java:624)
E/AndroidRuntime(11283):        at com.google.zxing.qrcode.detector.FinderPatternFinder.find(FinderPatternFinder.java:164)
E/AndroidRuntime(11283):        at com.google.zxing.qrcode.detector.Detector.detect(Detector.java:81)        
E/AndroidRuntime(11283):        at com.google.zxing.qrcode.QRCodeReader.decode(QRCodeReader.java:77)
E/AndroidRuntime(11283):        at com.google.zxing.MultiFormatReader.decodeInternal(MultiFormatReader.java:173)
E/AndroidRuntime(11283):        at com.google.zxing.MultiFormatReader.decodeWithState(MultiFormatReader.java:87)
E/AndroidRuntime(11283):        at com.journeyapps.barcodescanner.Decoder.decode(Decoder.java:75)
E/AndroidRuntime(11283):        at com.journeyapps.barcodescanner.Decoder.decode(Decoder.java:49)
E/AndroidRuntime(11283):        at com.journeyapps.barcodescanner.DecoderThread.decode(DecoderThread.java:146)
E/AndroidRuntime(11283):        at com.journeyapps.barcodescanner.DecoderThread.access$000(DecoderThread.java:22)
E/AndroidRuntime(11283):        at com.journeyapps.barcodescanner.DecoderThread$1.handleMessage(DecoderThread.java:38)
E/AndroidRuntime(11283):        at android.os.Handler.dispatchMessage(Handler.java:98)
E/AndroidRuntime(11283):        at android.os.Looper.loop(Looper.java:160)
E/AndroidRuntime(11283):        at android.os.HandlerThread.run(HandlerThread.java:61)
D/CameraPreview(11283): pause()
D/CameraInstance(11283): Closing camera

</p>
</details>

**Comment by juliansteenbakker** [comment]
Thank you all for reporting. I think updating ZXING broke the framework for android 5 and 6. I will try some things but for now please revert to version 0.5.1.

**Comment by dev-sandrosantos** [comment]
I'm having this same problem

**Comment by svprdga** [comment]
> Thank you all for reporting. I think updating ZXING broke the framework for android 5 and 6. I will try some things but for now please revert to version 0.5.1.

@juliansteenbakker Just to clarify, does reverting to `0.5.1` prevent this issue from happening?

**Comment by SainkP4G** [comment]
I've just tried with version 0.5.1 and get the same exception in Android 5.1.1

**Comment by svprdga** [comment]
I've just uploaded a hotfix with this downgrade...I'll report back if works for me...

**Comment by svprdga** [comment]
I can confirm that the issue is present also in version `0.5.1` :(

**Comment by svprdga** [comment]
> I can confirm that the issue is present also in version `0.5.1` :(

I was wrong, reverting to `0.5.1` **does fix the issue**.

Just make sure to add it the following way to pubspec.yaml: `qr_code_scanner: 0.5.1`

Because if you have it this way: `qr_code_scanner: ^0.5.1` when you do a `pub get` you will be downloading version `0.5.2`, which is the one with the issue.

**Comment by lhalcyon** [comment]
> > I can confirm that the issue is present also in version `0.5.1` :(
> 
> I was wrong, reverting to `0.5.1` **does fix the issue**.
> 
> Just make sure to add it the following way to pubspec.yaml: `qr_code_scanner: 0.5.1`
> 
> Because if you have it this way: `qr_code_scanner: ^0.5.1` when you do a `pub get` you will be downloading version `0.5.2`, which is the one with the issue.

Thx , u save the day !

**Comment by tehsunnliu** [comment]
Hi, I just discovered this bug, and it crashes the whole app. Even while running on a real device while debugging.

```
E/AndroidRuntime(  891): FATAL EXCEPTION: DecoderThread
E/AndroidRuntime(  891): Process: com.example, PID: 891
E/AndroidRuntime(  891): java.lang.NoSuchMethodError: No interface method sort(Ljava/util/Comparator;)V in class Ljava/util/List; or its super classes (declaration of 'java.util.List' appears in /system/framework/core-libart.jar)
E/AndroidRuntime(  891): 	at com.google.zxing.qrcode.detector.FinderPatternFinder.selectBestPatterns(FinderPatternFinder.java:624)
E/AndroidRuntime(  891): 	at com.google.zxing.qrcode.detector.FinderPatternFinder.find(FinderPatternFinder.java:164)
E/AndroidRuntime(  891): 	at com.google.zxing.qrcode.detector.Detector.detect(Detector.java:81)
E/AndroidRuntime(  891): 	at com.google.zxing.qrcode.QRCodeReader.decode(QRCodeReader.java:77)
E/AndroidRuntime(  891): 	at com.google.zxing.MultiFormatReader.decodeInternal(MultiFormatReader.java:173)
E/AndroidRuntime(  891): 	at com.google.zxing.MultiFormatReader.decodeWithState(MultiFormatReader.java:87)
E/AndroidRuntime(  891): 	at com.journeyapps.barcodescanner.Decoder.decode(Decoder.java:75)
E/AndroidRuntime(  891): 	at com.journeyapps.barcodescanner.Decoder.decode(Decoder.java:49)
E/AndroidRuntime(  891): 	at com.journeyapps.barcodescanner.DecoderThread.decode(DecoderThread.java:146)
E/AndroidRuntime(  891): 	at com.journeyapps.barcodescanner.DecoderThread.access$000(DecoderThread.java:22)
E/AndroidRuntime(  891): 	at com.journeyapps.barcodescanner.DecoderThread$1.handleMessage(DecoderThread.java:38)
E/AndroidRuntime(  891): 	at android.os.Handler.dispatchMessage(Handler.java:98)
E/AndroidRuntime(  891): 	at android.os.Looper.loop(Looper.java:157)
E/AndroidRuntime(  891): 	at android.os.HandlerThread.run(HandlerThread.java:61)
D/CameraPreview(  891): pause()

``` 
This happens only in my older device running Android 29 (6.0.1 MMB29M)
However, it works perfectly fine on my new device.

> > I can confirm that the issue is present also in version `0.5.1` :(
> 
> I was wrong, reverting to `0.5.1` **does fix the issue**.
> 
> Just make sure to add it the following way to pubspec.yaml: `qr_code_scanner: 0.5.1`
> 
> Because if you have it this way: `qr_code_scanner: ^0.5.1` when you do a `pub get` you will be downloading version `0.5.2`, which is the one with the issue.

This Fixed the issue thanks.

**Comment by m-ochyra** [comment]
"Platform Version to low for camera permission check" error is fixed in PR #450
